File: typr-tester-typr-java/src/java/adventureworks/update_person_returning/UpdatePersonReturningSqlRepoTest.java
package adventureworks.update_person_returning;

import adventureworks.DbNow;
import adventureworks.WithConnection;
import java.util.Optional;
import org.junit.Test;

public class UpdatePersonReturningSqlRepoTest {
  private final UpdatePersonReturningSqlRepoImpl updatePersonReturningSqlRepo =
      new UpdatePersonReturningSqlRepoImpl();

  @Test
  public void timestampWorks() {
    WithConnection.run(
        c -> {
          updatePersonReturningSqlRepo.apply(
              Optional.of("1"), Optional.of(DbNow.localDateTime()), c);
        });
  }
}

File: typr-tester-typr-scala-old/src/scala/adventureworks/update_person_returning/UpdatePersonReturningSqlRepoTest.scala
package adventureworks.update_person_returning

import adventureworks.{DbNow, WithConnection}
import org.junit.Test

import java.util.Optional

class UpdatePersonReturningSqlRepoTest {
  private val updatePersonReturningSqlRepo = new UpdatePersonReturningSqlRepoImpl

  @Test
  def timestampWorks(): Unit = {
    WithConnection {
      val _ = updatePersonReturningSqlRepo.apply(Optional.of("1"), Optional.of(DbNow.localDateTime()))
    }
  }
}
